plugins/messagelayer/consensus_plugin.go

```diff
@@ -225,7 +226,9 @@ type OpinionGivers map[identity.ID]OpinionGiver
 // Query retrieves the opinions about the given conflicts and timestamps.
 func (o *OpinionGiver) Query(ctx context.Context, conflictIDs []string, timestampIDs []string) (opinions opinion.Opinions, err error) {
 	for i := 0; i < StatementParameters.WaitForStatement; i++ {
-		if o.view != nil {
+		// query statement only if it's been received within the last round interval
+		// otherwise node might be down and we don't want to look at old statements in subsequent rounds
```

Comment:
see above. basically, if the node has been active in this round, we should consider the last statement opinion about a conflict of the node, even if the opinion is already 10 rounds old. this is because the node might have finalized in FPC and doesn't chat about a certain conflict anymore - yet we have to consider its last broadcast opinion.

Reply:
so we have to make sure that the statement opinions dont get emptied, and will be considered in subsequent rounds (if the node is still active)

Reply:
That's how it works - conflicts are removed from the statement registry some time after last arrival (5 minutes by default), when the voting is resolved. 
I might have not described it correctly, but it works as you described - in this improvement we check if the node received a statement within last 10 seconds and it works independently of opinion cleaning in registry.